_additional/frontend/front.js

```diff
@@ -0,0 +1,130 @@
+                            } 
+                            var elem = document.getElementById('messages');
+                            elem.scrollTop = elem.scrollHeight;                            
+                        }                    
+                    }
+                });
+            }            
+            
+           function refreshMessages () {
+               setTimeout(function() {
+                   showMessages();
+                   getUsers();
+                   refreshMessages();                   
+               }, 500);
+           }
+
+            function sendMessage (e) {
+                e.preventDefault(); 
+                var messageBody = $('#messageBody').val();  
+                var message = {
+                    body: messageBody,
+                    username: name
+                };             
+                $.ajax({
+                    type: 'POST',
+                    url: '/messages',
+                    dataType: 'JSON',
+                    data: message,
+                    success: function(response) {
+                        $('#messageBody').val('');
+                    },
+                    error: function(response) {
+                        console.log(JSON.stringify(response));
+                    }
+                });
+            }            
+            
+            function login(e) {                
+                e.preventDefault();
+                name = $('#nameInput').val();
+                $.ajax({
+                    type: 'POST',
+                    url: '/connect',
+                    dataType: 'JSON',
+                    data: { 'username': name },
+                    success: function(response) {
+                        showMessages();
+                        $('#nameInput').val('');
+                        $('#nameInput').attr('disabled', true);
+                        $('#nameSubmit').attr('disabled', true);
+                        $('#messageBody').attr('disabled', false);
+                        $('#sendButton').attr('disabled', false);
+                    },
+                    error: function(response) {
+                        alert(JSON.stringify(response.responseJSON.error));
+                    }
+                });
+            }
+
+            function logout(e) {
+                e.preventDefault();
+                $.ajax({
+                    type: 'POST',
+                    url: '/disconnect',
+                    dataType: 'JSON',
+                    data: { 'username': name },
+                    success: function(response) {
+                        $('#nameInput').val('');
+                        $('#nameInput').attr('disabled', false);
+                        $('#nameSubmit').attr('disabled', false);
+                        $('#messageBody').attr('disabled', true);
+                        $('#sendButton').attr('disabled', true);
+                        refreshMessages();

```

Comment:
To tu jest potrzebne? Czy to nam nie spowoduje uruchomienia jakby drugiej instancji refreshera?